include/semaphore/semaphore.hpp

```diff
@@ -0,0 +1,51 @@
+/**
+ * @file semaphore.hpp
+ * 
+ * @brief Generic semaphore class header
+ * 
+ * @date 06/2024
+ */
+
+#ifndef __SEMAPHORE_HPP__
+#define __SEMAPHORE_HPP__
+
+#include <Arduino.h>
+#include <freertos/FreeRTOS.h>
+#include <freertos/semphr.h>
+
+namespace rfidoor::semaphore {
+
+class Semaphore {
+public:
+  /**
+   * @brief Constructor for the generic Semaphore class
+   */
+  Semaphore(void);
+
+  /**
+   * @brief Create semaphore function to be overriden by the derived class
+   * 
+   * @return SemaphoreHandle_t Handle for the semaphore
+   */
+  virtual SemaphoreHandle_t create_semaphore() { return NULL; };
```

Comment:
Aqui nos outros lugares eu tava fazendo tipo
```suggestion
  virtual SemaphoreHandle_t create_semaphore() = 0;
```

Não sabia que dava pra fazer com o `return NULL`, achei dahora